### PR TITLE
fixing myst nb execution bug

### PR DESCRIPTION
**Pull request by choldgraf**
This fixes https://github.com/executablebooks/MyST-NB/issues/152

If it doesn't find the file extension information, it will raise a warning and choose `.py` as a default. That way we still generate the notebook etc, it just won't have outputs.

### CODE REVIEW

File: myst_nb/parser.py [hunk 1]
```diff
@@ -293,6 +293,14 @@ def nb_output_to_disc(ntbk: nbf.NotebookNode, document: nodes.document) -> Path:
     doc_filename = path_doc.name
     build_dir = Path(document.settings.env.app.outdir).parent
     output_dir = build_dir.joinpath("jupyter_execute", doc_relpath)
+
+    # Write a script too.
+    if not ntbk.metadata.get("language_info"):
+        ntbk.metadata["language_info"] = {"file_extension": ".py"}
```
Comment (chrisjsewell) [inline myst_nb/parser.py:299]: Just use `.txt`, to account for when the notebook kernel is not a python one?

**Comment by chrisjsewell** [comment]
Note I have also opened a PR upstream jupyter/jupyter-sphinx#127

File: myst_nb/parser.py (hunk 1 above)
Comment (choldgraf) [inline myst_nb/parser.py:299]:
```suggestion
        ntbk.metadata["language_info"] = {"file_extension": ".txt"}
```

File: myst_nb/parser.py [hunk 2]
```diff
@@ -293,6 +293,14 @@ def nb_output_to_disc(ntbk: nbf.NotebookNode, document: nodes.document) -> Path:
...
+        SPHINX_LOGGER.warning(
+            "Notebook code has no file extension metadata, " "defaulting to `.py`",
```
Comment (choldgraf) [inline myst_nb/parser.py:301]:
```suggestion
            "Notebook code has no file extension metadata, " "defaulting to `.txt`",
```

**Comment by choldgraf** [comment]
good call @chrisjsewell - I changed to `.txt` (and anyway I guess we will remove this code once jupyter-sphinx gets in your PR)

File: myst_nb/parser.py [hunk 3]
```diff
@@ -293,6 +293,14 @@ def nb_output_to_disc(ntbk: nbf.NotebookNode, document: nodes.document) -> Path:
     doc_filename = path_doc.name
     build_dir = Path(document.settings.env.app.outdir).parent
     output_dir = build_dir.joinpath("jupyter_execute", doc_relpath)
+
+    # Write a script too.
+    if not ntbk.metadata.get("language_info"):
+        ntbk.metadata["language_info"] = {"file_extension": ".txt"}
```
Comment (choldgraf) [inline myst_nb/parser.py:299]:
```suggestion
        # TODO: we can remove this
        # once https://github.com/executablebooks/MyST-NB/issues/177 is merged
        ntbk.metadata["language_info"] = {"file_extension": ".txt"}
```

**Comment by choldgraf** [comment]
OK I'm gonna merge this once since it fixes a nasty bug, and we can remove when the upstream patch is in

**Pull request merged**